hera_corr_cm/redis_cm.py

```diff
@@ -0,0 +1,142 @@
+    redishost.hmset('corr:map', redhash)
+
+
+def hera_antpol_to_ant_pol(antpol):
+    """Parse antpol."""
+    antpol = antpol.lower().lstrip('h')
+    pol = antpol[-1]
+    ant = int(antpol[:-1])
+    return ant, pol
+
+
+def get_snaps_from_redis():
+    """Read SNAPs from redis - from CM, config and correlator viewpoints."""
+    import yaml
+
+    r = redis.Redis('redishost')
+    snaps_cm_list = list(json.loads(r.hget('corr:map', 'all_snap_inputs')).keys())
+    snap_to_host = json.loads(r.hget('corr:map', 'snap_host'))
+    snaps = {'cm': [], 'cfg': [], 'corr': []}
+    for snap in snaps_cm_list:
+        try:
+            snaps['cm'].append(snap_to_host[snap])
+        except KeyError:
+            continue
+    snaps['cfg'] = list(yaml.safe_load(r.hget('init_configuration', 'config'))['fengines'].keys())
+    snaps['corr'] = list(r.hgetall('corr:snap_ants').keys())
+    return snaps
+
+
+def read_maps_from_redis(redishost='redishost'):
+    """Read subset of corr:map."""
+    if isinstance(redishost, str):
+        redishost = redis.Redis(redishost)
+    if not redishost.exists('corr:map'):
+        return None
+    x = redishost.hgetall('corr:map')
+    x['update_time'] = float(x['update_time'])
+    x['ant_to_snap'] = json.loads(x['ant_to_snap'])
+    x['snap_to_ant'] = json.loads(x['snap_to_ant'])
+    return x
+
+
+def read_cminfo_from_redis(return_as, redishost='redishost'):
+    """
+    Read location information from redis.
+
+    This is a redis replacement for the cminfo methods that used hera_mc.  For
+    "drop-in" replacement in how it gets used in various places there are two
+    different return options per the "return_as" parameter.
+
+    Parameters
+    ----------
+    return_as : str
+                If return_as 'dict': returns dict as per old cminfo
+                If return_as 'namespace':  returns as a Namespace for other places
+    redishost : str
+                Name of redis host.
+
+    Returns
+    -------
+    dict (or Namespace) of cminfo
+    """
+    cminfo = {'antenna_numbers': None, 'antenna_names': None,               # Make identical to cminfo  # noqa
+              'antenna_positions': None, 'antenna_alts': None,              # returned by hera_mc
+              'antenna_utm_eastings': None, 'antenna_utm_northings': None,  # These are same keys
+              'correlator_inputs': None, 'cm_version': None}
+    redishost = redis.Redis('redishost')
+    for k in cminfo.keys():
+        cminfo[k] = json.loads(redishost.hget('corr:map', k))
+    cofa_info = json.loads(redishost.hget('corr:map', 'cofa'))
+    cminfo['cofa_lat'] = cofa_info['lat']
+    cminfo['cofa_lon'] = cofa_info['lon']
+    cminfo['cofa_alt'] = cofa_info['alt']
+    cofa_xyz = json.loads(redishost.hget('corr:map', 'cofa_xyz'))
+    cminfo['cofa_X'] = cofa_xyz[0]
+    cminfo['cofa_Y'] = cofa_xyz[1]
+    cminfo['cofa_Z'] = cofa_xyz[2]
+
+    if not return_as.lower().startswith('d'):
```

Comment:
While this will "work" I would be more comfortable if the input checks whether `return_as` is exclusively in the accepted parameters. something like checking `return_as.lower() in ["namespace", "dict']` at the start of the function. Otherwise we could input `dummy` and still get the dict output.